src/examples/learn_quant/scripts/generate_expressions.py

```diff
@@ -0,0 +1,171 @@
+from yaml import dump
+from time import time
+import argparse
+from copy import deepcopy
+import pickle
+import os
+import csv
+import pathlib
+from pathlib import Path
+
+try:
+    from yaml import CDumper as Dumper
+except ImportError:
+    from yaml import Dumper
+
+import hydra
+from omegaconf import DictConfig, OmegaConf
+
+from ultk.language.grammar import GrammaticalExpression
+from typing import Any
+
+from ..quantifier import QuantifierUniverse
+from ..grammar import QuantifierGrammar
+from ..meaning import create_universe
+from ..util import save_quantifiers, save_inclusive_generation
+
+def enumerate_quantifiers(depth: int, 
+                          quantifiers_universe: QuantifierUniverse, 
+                          quantifiers_grammar: QuantifierGrammar
+                        ) -> dict[GrammaticalExpression, Any]:
+
+    expressions_by_meaning = quantifiers_grammar.get_unique_expressions(
+        depth,
+        max_size=2 ** len(quantifiers_universe),
+        unique_key=lambda expr: expr.evaluate(quantifiers_universe),
+        compare_func=lambda e1, e2: len(e1) < len(e2),
+    )
+
+    # filter out the trivial meaning, results in NaNs
+    # iterate over keys, since we need to change the dict itself
+    for meaning in list(expressions_by_meaning.keys()):
+        if len(meaning.referents) == 0:
+            del expressions_by_meaning[meaning]
+    
+    return expressions_by_meaning
+
+def generate_expressions(quantifiers_grammar: QuantifierGrammar, 
```

Comment:
Please run `black` for formatting too :).  The repo is setup to do that automatically when PRs are merged, but I'm having a hard time reading this method actually